### ISSUE TITLE
Future of generator-backbone

### ISSUE THREAD

**Issue by ruyadorno** [yeoman/generator-backbone#385]
As one of the original Yeoman generators, **generator-backbone** has a lot of baggage and so many options/opinions that makes it a heavy burden to maintain - as a result the generator lost contributors and went through a poorly maintained phase. For such an iconic generator, it would be a shame to end up abandoned and filled with unresolved issues.

Here are some ideas for a better future:
- [ ] Reshape **generator-backbone** as a composable foundation for other generators
- [ ] Extract all opinions/choices into new separated generators that builds upon this one
- [ ] Create a new major release `v1.0.0` for this new clear slate - this way we can still have `v0.4.x` around for anyone depending on it

I'm writing this down with the goal of collecting feedback and overall opinion from interested people - personally, I think this can be a good opportunity to set a good example on how to create/maintain composable generators, similar to what the Yeoman team did with `generator-node` -> `generator-generator`.

cc/ people that were involved with **generator-backbone** in the past: @SBoudrias @addyosmani @revathskumar

**Comment by ruyadorno** [comment]
Also cc/ people that have been somehow involved with the project lately: @marian-r @mteodori @tconroy @arthurvr

**Comment by sindresorhus** [comment]
:+1:

**Comment by marian-r** [comment]
Yeah, awesome idea. I think this generator should show the best practices for modern web app development in Backbone.js and maybe provide a foundation for other generators.

**Comment by rodfersou** [comment]
:+1:

**Comment by mteodori** [comment]
great idea, happy to help!

**Comment by rodfersou** [comment]
need to break the ideas in small issues

**Comment by ruyadorno** [comment]
@rodfersou good idea, once I start working on this I'll create one issue to track progress on `v1.0.0` and others for each of the flavors

for now I'll keep updating progress here to keep you guys posted :blush:

**Comment by rodfersou** [comment]
:+1:

**Comment by mteodori** [comment]
I've been creating some PRs for some of the latest bugs but I still struggle on composable generators a bit, for example when a command line option/prompt is set by a composed generator like "ui", the mocha ui style from generator-mocha, how can I re-use it in another generator when the storage of this value is namespaced by generator name in .yo-rc.json? I ended up loading it explicitly from that file but feels like an hack

**Comment by mischah** [comment]
Opps. Just stumbled upon this in the context of https://github.com/yeoman/yeoman/issues/1653

Now I would love to know if there is a chance to keep generator-backbone alive.

**Comment by marian-r** [comment]
@ruyadorno what are your thoughts on this? I guess the most important question here is if there is anybody with a room for contributions?

I checked npm and there are 800 monthly installs of the generator. Do you think the number is low? And if so, is it low because of the current state of the generator? So, if we improve it would get higher traction? I know that days of Backbone might be a bit over, but I'm sure there are people willing use it if it fits their needs. Backbone has 700K monthly installs on npm.

**Comment by ruyadorno** [comment]
@marian-r totally agree, would be a shame to deprecate such a historically important generator, I jumped in to help at a time the generator was completely broken but unfortunately I don't have the time to move on with the outlined plans 😞  sorry by failing to update you guys on that, almost a year went by since my last update 🙀 

@mischah essentially we just need someone to carry forward the original plan of rewriting a new `v1.0.0` containing only the backbone parts and removing the dozen of setup options - the more tricky part is moving on with the second part of the plan (getting other generators to fill the gap of providing these dozen of options and maintainers for all those of course)

I'm happy to keep helping with code reviews and small fixes

**Comment by mteodori** [comment]
I am fine to step in, I am using Backbone at work with Marionette but it would be nice to understand which of the current options are still relevant, my current stack is on ES6 with Babel, Handlebars Sass and webpack2 with some extra help from grunt

**Comment by SBoudrias** [comment]
@mteodori I think that's a pretty good setup. We definitively would want to move to webpack + es6. Simplify the whole build process and make the generator leaner.

**Comment by mischah** [comment]
Great 💖 

Just added generator-backbone to the to do list [regarding issue and PR templates](https://github.com/yeoman/yeoman/issues/1652) 😘

**Comment by roopeshchinnakampalli** [comment]
Tests are breaking with this generator. May be happening with other generators too!
As per the articles on the web, Phantom JS 2.x & above are compatible on macOS Sierra. We may have to fix this.

Is anyone working actively on this generator?

**Comment by SBoudrias** [comment]
@roopeshreddy no one is actively maintaining this generator.

**Comment by ruyadorno** [comment]
I'm still around if anyone is willing to submit a PR but def not actively working on it
